src/main/java/sirius/biz/storage/layer3/VirtualFile.java

```diff
@@ -0,0 +1,794 @@
+     *
+     * @throws HandledException if the file cannot be renamed
+     */
+    public void rename(String newName) {
+        if (!tryRename(newName)) {
+            throw Exceptions.createHandled().withNLSKey("VirtualFile.cannotRename").set("file", path()).handle();
+        }
+    }
+
+    /**
+     * Determines if this file can (probably) be move in an efficient way.
+     *
+     * @return <tt>true</tt> if this file can be efficiently moved or <tt>false</tt> otherwise
+     */
+    public boolean canFastMove() {
+        try {
+            if (canMoveHandler != null) {
+                return canMoveHandler.test(this);
+            }
+
+            return moveHandler != null;
+        } catch (Exception e) {
+            throw handleErrorInCallback(e, "canMoveHandler");
+        }
+    }
+
+    /**
+     * Tries to efficiently move the file in the given directory.
+     *
+     * @param newParent the new parent directory
+     * @return <tt>true</tt> if the operation was successful, <tt>false</tt> otherwise
+     */
+    public boolean tryFastMoveTo(VirtualFile newParent) {
+        try {
+            if (moveHandler == null) {
+                return false;
+            }
+
+            return moveHandler.apply(this, newParent);
+        } catch (Exception e) {
+            throw handleErrorInCallback(e, "moveHandler");
+        }
+    }
+
+    /**
+     * Efficiently moves the file in the given directory.
+     *
+     * @param newParent the new parent directory
+     * @throws HandledException if the file cannot be efficiently moved
+     */
+    public void fastMoveTo(VirtualFile newParent) {
+        if (!tryFastMoveTo(newParent)) {
+            throw Exceptions.createHandled().withNLSKey("VirtualFile.cannotMove").set("file", path()).handle();
+        }
+    }
+
+    /**
+     * Determines if this file can bei moved (either via an efficient implementation or by copy+delete).
+     *
+     * @return <tt>true</tt> if the file can (probably) be moved, <tt>false</tt> otherwise
+     */
+    public boolean canMove() {
+        return canFastMove() || canDelete();
+    }
+
+    /**
+     * Tries to move this file in the given directory.
+     * <p>
+     * Note that this might be a long running operation if an efficient implementation cannot be used.
+     *
+     * @param newParent the new parent directory
+     * @return <tt>true</tt> if the operation was successful, <tt>false</tt> otherwise
+     */
+    public boolean tryMoveTo(VirtualFile newParent) {
+        if (tryFastMoveTo(newParent)) {
+            return true;
+        }
+
+        //TODO implement move manually
+        throw new UnsupportedOperationException("Move is currently not implemented");
+    }
+
+    /**
+     * Determines if the file exists.
+     *
+     * @return <tt>true</tt> if the file exists, <tt>false</tt> otherwise
+     */
+    public boolean exists() {
+        try {
+            if (existsFlagSupplier == null) {
+                return false;
```

Comment:
I think inverting this if check would improve readability/flow and would be consistent to the other methods in this class